### DISCUSSION TITLE
Error in mesh the domain in example 1

### DISCUSSION THREAD

**Discussion by fei1127-CQ** [neperfepx/FEPX#13] [Q&A] [answered]
Hi, 
After install the Neper (4.1.2) and FEPX (1.1.1), I try to repeat the example 1 in FEPX. I bulit a simulation.tess through the code in example 1.However,  an error appeared during the mesh of the domain as follow:
```
gof@DESKTOP-9E3S5B6:/mnt/d/CPFM/test$ neper -M simulation.tess -order 2 -rcl 1.25 -part 2
========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.1.2
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt
Info   : Running on 12 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2020, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/gof/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -M loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] simulation.tess -order 2 -rcl 1.25 -part 2
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   :   - Reading arguments...
Error  : Options `-part*' (mesh partitioning) are not available since
         this version of Neper has been built without libscotch.
Aborted (core dumped)
```

Which version of Neper can be used for this simulation? or is there any other way to solve this problem?

**Comment by rquey** [comment]
You have to include libscotch at Neper's configuration stage - see the [PDF documentation](https://neper.info/docs/neper.pdf), Section 1.2.

Then you'll see this in Neper's header:
```
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
```

**Reply by mkasemer** [reply]
Indeed, this is an error in Neper due to the fact that you have not built the program with `libscotch`. For any issues with the program Neper (rather than FEPX), please post in the [Neper discussion forum](https://github.com/rquey/neper/discussions).

**Reply by fei1127-CQ** [reply]
Thanks for all your helps. I just make it.

**Comment by dannypanduro** [comment, marked as answer]
Hi fei1127-CQ, 

I had a similar issue with not building Neper with libscotch. I solved this by the following steps below.

1.) Make sure you have installed the libscotch dependency. You can do this by typing the following command in your terminal: 
$ apt-get install libscotch-dev

I use Ubuntu. If you do too, you would need to type:
$ sudo apt-get install libscotch-dev

2.) Start installing Neper.  The key is you need to type "$ cmake .. -DHAVE_LIBCOTCH=ON" instead of just "$ cmake .."

Download and extract Neper source code then:
$ cd neper-4.0.2/src
$ mkdir build
$ cd build
$ cmake .. -DHAVE_LIBSCOTCH=ON
$ make
$ make install

If you used Ubuntu like me, then the code would be:
$ cd neper-4.0.2/src
$ mkdir build
$ cd build
$ cmake .. -DHAVE_LIBSCOTCH=ON
$ sudo make
$ sudo make install

I hope this helps! Cheers!

Richard

**Reply by rquey** [reply]
Yep. FYI, the next Neper version will include libscotch in its source (but still use the libscotch installed system-wide if available) to make things simpler for everyone here...

**Reply by mkasemer** [reply]
@RichardPA (and others): For future reference, issues for dependencies are covered in the [Neper documentation](https://neper.info/docs/neper.pdf). As @rquey mentioned in the previous comment, future Neper releases may not require "manual" installation of libscotch and other dependencies, so the documentation is the best source of up-to-date information.

Regarding the usage of `sudo` (for both Neper and FEPX): Indeed, the sudo command may be necessary when root access is required, though generally this may not be the case. Use when necessary. As a note: installation on shared systems (workstations, clusters) may require root access that is not granted to the general user, and thus you can either 1) request a system administrator to handle the installation or 2) move the `neper` or `fepx` to a binary directory located in your specific home directory which you would want to add to your path (e.g., `/home/<username>/bin`). In the latter case, this binary will likely be accessible only to the user who installed it / has it in their directory.

Again, any issues regarding Neper should be directed to the [Neper discussion forum](https://github.com/rquey/neper/discussions).

**Reply by dannypanduro** [reply]
Good to hear about the next Neper version. I'm sure that'll be helpful! 

@mkasemer, thank you for the explanation on sudo! That makes sense.

**Reply by fei1127-CQ** [reply]
Thanks for all your helps. I just make it.

**Reply by fei1127-CQ** [reply]
> 
> 
> Yep. FYI, the next Neper version will include libscotch in its source (but still use the libscotch installed system-wide if available) to make things simpler for everyone here...

Waiting for your new version. The program is quite strengthful.
